### PR TITLE
add delete endpoint for pluto commission deletion

### PR DESCRIPTION
**Pull request by akash1810**
deleting a commission will first delete all the projects associated with it

### CODE REVIEW

**Comment by akash1810** [comment]
cc @fredex42 @yusufparkar

File: app/controllers/PlutoController.scala [hunk 1]
```diff
@@ -37,6 +37,12 @@ class PlutoController(
     Ok(Json.toJson(plutoProjects))
   }
 
+  def deleteCommission(id: String) = APIHMACAuthAction {
+    stores.plutoProjectStore.deleteByCommissionId(id)
```
Comment (mbarton) [inline app/controllers/PlutoController.scala]: Should we check the return result of these two to catch errors?

File: common/src/main/scala/com/gu/media/pluto/PlutoProjectDataStore.scala [hunk 2]
```diff
@@ -43,4 +43,9 @@ class PlutoProjectDataStore(aws: DynamoAccess, plutoCommissionDataStore: PlutoCo
     Scanamo.exec(aws.dynamoDB)(op)
     project
   }
+
+  def deleteByCommissionId(commissionId: String) = {
+    log.info(s"deleting all pluto projects for commission $commissionId")
+    getByCommissionId(commissionId).map(project => Scanamo.delete(aws.dynamoDB)(table.name)('id -> project.id))
```
Comment (mbarton) [inline common/src/main/scala/com/gu/media/pluto/PlutoProjectDataStore.scala]: as above, do we know if one of the deletes failed?

File: app/controllers/PlutoController.scala (hunk 1 above)
Comment (akash1810) [reply app/controllers/PlutoController.scala]: `Scanamo.delete` returns a `DeleteItemResult`, which, looking at [the docs](https://docs.aws.amazon.com/sdkfornet1/latest/apidocs/html/T_Amazon_DynamoDB_Model_DeleteItemResult.htm), returns on success or failure. I don't think we necessarily care if we try to delete a non-existing commission...

Comment (mbarton) [reply app/controllers/PlutoController.scala]: Who re-sends the message?

Comment (akash1810) [reply app/controllers/PlutoController.scala]: Pluto would.

Comment (mbarton) [reply app/controllers/PlutoController.scala]: OK cool, so it checks to see if the deletion has actually worked and retries accordingly?

**Pull request merged**